scio-extra/src/main/scala/com/spotify/scio/extra/voyager/package.scala

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2023 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.scio.extra
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.annotations.experimental
+import com.spotify.scio.values.{SCollection, SideInput}
+import com.spotify.voyager.jni.Index.{SpaceType, StorageDataType}
+import com.spotify.voyager.jni.StringIndex
+import org.apache.beam.sdk.transforms.{DoFn, View}
+import org.apache.beam.sdk.values.PCollectionView
+import org.slf4j.LoggerFactory
+
+import java.util.UUID
+import scala.collection.mutable
+
+/** Main package for Voyager side input APIs. Import all. */
+package object voyager {
+  sealed abstract class VoyagerDistanceMeasure
+  case object Euclidean extends VoyagerDistanceMeasure
+  case object Cosine extends VoyagerDistanceMeasure
+  case object Dot extends VoyagerDistanceMeasure
+
+  sealed abstract class VoyagerStorageType
+  case object Float8 extends VoyagerStorageType
+  case object Float32 extends VoyagerStorageType
+  case object E4M3 extends VoyagerStorageType
```

Comment:
Is there a reason we do not use the java values ?

Reply:
I went to leverage a similar pattern to how Annoy put a facade over the internal library values in favor of internal case object ones. I'm more than happy to move these back to just accepting the Voyager ones directly.